src/hex_api_release.erl

```diff
@@ -79,6 +80,41 @@ publish(Config, Tarball) when is_map(Config) and is_binary(Tarball) ->
     Body = {TarballContentType, Tarball},
     hex_api:post(Config2, Path, Body).
 
+%% @doc
+%% Replaces an existing package release.
+%%
+%% Examples:
+%%
+%% ```
+%% > hex_api_release:replace(hex_core:default_config(), Tarball).
+%% {ok, {201, ..., #{
+%%      <<"checksum">> => <<"540d210d81f56f17f64309a4896430e727972499b37bd59342dc08d61dff74d8">>,
+%%      <<"docs_html_url">> => <<"https://hexdocs.pm/package/1.0.0/">>,
+%%      <<"downloads">> => 740,<<"has_docs">> => true,
+%%      <<"html_url">> => <<"https://hex.pm/packages/package/1.0.0">>,
+%%      <<"inserted_at">> => <<"2014-12-09T18:32:03Z">>,
+%%      <<"meta">> =>
+%%          #{<<"app">> => <<"package">>,
+%%            <<"build_tools">> => [<<"mix">>]},
+%%      <<"package_url">> => <<"https://hex.pm/api/packages/package">>,
+%%      <<"publisher">> => nil,<<"requirements">> => #{},
+%%      <<"retirement">> => nil,
+%%      <<"updated_at">> => <<"2019-07-28T21:12:11Z">>,
+%%      <<"url">> => <<"https://hex.pm/api/packages/package/releases/1.0.0">>,
+%%      <<"version">> => <<"1.0.0">>
+%%      }}}
+%% '''
+%% @end
+-spec replace(hex_core:config(), binary()) -> hex_api:response().
+replace(Config, Tarball) when is_map(Config) and is_binary(Tarball) ->
```

Comment:
so my initial though we'd have `publish(Config, Tarball, [{replace, true}])` (and in the future other potential options) but I think making this a separate explicit operation is a good idea too. Do you have a preference @ericmj?

Reply:
I think an option on `publish` is preferable. But regardless of this we need `publish` to default to `?replace=false`.